### DISCUSSION TITLE
Is it possible to allow a custom Bash script as a post-upgrade command?

### DISCUSSION THREAD

**Discussion by chopeen** [renovatebot/renovate#25552] [Suggest an Idea]
A user requested being able to run their custom script (some text manipulation with `grep`, `awk`, `gsed`) as a post-upgrade command. The goal is to extract some version number from a `*.tf` file, increment it and save to the same file. Supposedly, there is no standard `terraform` command for this operation (I am not a Terraform expert, so I can neither confirm it or deny it).

What's a secure way to satisfy this request?

### Possible solution

I run Renovate using a custom image built from `renovate/renovate:full`, so I could be adding this script to `/usr/local/bin` when building the image and then allow `/usr/local/bin/custom.sh` in the `allowedPostUpgradeCommands` setting.

Could this work?
Are there any obvious downsides I am not seeing?

**Comment by rarkins** [comment]
When you say "secure", who are you securing against? If you trust your developers/users then all commands are safe to allow.

**Reply by chopeen** [reply]
I trust my colleagues, but I still prefer to have a limited number of allowed commands.

Also, expressing the entire script (partial example below) as a regex won't be fun:

```
version=$(grep -o 'version[[:space:]]*=[[:space:]]*"[^"]*"' main.tf | awk -F '"' '{print $2}')
major=$(echo $version | cut -d. -f1)

[...]

gsed -i "s/version[[:space:]]*=[[:space:]]*\"$version\"/version = \"$new_version\"/" main.tf
```

**Reply by rarkins** [reply]
I think your proposed solution could work
